bluemira/power_cycle/tools.py

```diff
@@ -0,0 +1,334 @@
+            )
+    return dictionary
+
+
+def validate_lists_to_have_same_length(*args):
+    """
+    Validate an arbitrary number of arguments to be lists of the same
+    length. Arguments that are not lists are considered lists of one
+    element.
+    If all equal, returns the length of those lists.
+    """
+    all_lengths = []
+    for argument in args:
+        argument = validate_list(argument)
+        argument_length = len(argument)
+        all_lengths.append(argument_length)
+    unique_lengths = unique_and_sorted_vector(all_lengths)
+
+    all_lenghts_are_not_equal = len(unique_lengths) != 1
+    if all_lenghts_are_not_equal:
+        raise ValueError(
+            "At least one of the lists passed as argument does not "
+            "have the same length as the others."
+        )
+    else:
+        return argument_length
+
+
+# ######################################################################
+# MANIPULATION
+# ######################################################################
+
+
+def copy_dict_without_key(dictionary, key_to_remove):
+    """
+    Returns a dictionary that is a copy of the parameter 'dictionary',
+    but without the 'key_to_remove' key.
+    """
+    d = dictionary
+    dictionary_without_key = {k: d[k] for k in d if k != key_to_remove}
+    return dictionary_without_key
+
+
+def unnest_list(list_of_lists):
+    """
+    Un-nest a list of lists into a simple list, maintaining order.
+    """
+    return [item for sublist in list_of_lists for item in sublist]
+
+
+def unique_and_sorted_vector(vector):
+    """
+    Returns a sorted list, in ascending order, created from the set
+    created from a vector, as a way to eliminate redundant entries.
+    """
+    vector = validate_vector(vector)
+    unique_vector = list(set(vector))
+    sorted_vector = sorted(unique_vector)
+    return sorted_vector
+
+
+def remove_characters(string, character_list):
+    """
+    Remove all 'str' in a list from a main 'string' parameter.
+    """
+    for character in character_list:
+        string = string.replace(character, "")
+    return string
+
+
+def convert_string_into_numeric_list(string):
+    """
+    Converts a string in the format '[x, y, z]' imported from a json
+    into a list on numbers.
+    """
+    string = copy.deepcopy(string)
+    unwanted_characters = [" ", "[", "]"]
+    clean_string = remove_characters(string, unwanted_characters)
+    list_from_string = list(clean_string.split(","))
+    numbers_list = [float(n) for n in list_from_string]
+    return numbers_list
+
+
+def read_json(file_path):
+    """
+    Returns the contents of a 'json' file in 'dict' format.
+    """
+    try:
+        with open(file_path) as json_file:
+            contents_dict = json.load(json_file)
+    except json.decoder.JSONDecodeError:
+        raise TypeError(
+            "The file could not be read as a 'json' file.",
+        )
+    return contents_dict
```

Comment:
I think a lot of things will be simplified if you do a bit more processing of the json files here. List of numbers shouldnt be stored as strings in the json eg this

```
                        "time": [
                            "[0, 1]",
                            "[0, 1]",
                            "[0, 1]"
                        ],
```

should just be this
```
                        "time": [
                            [0, 1],
                            [0, 1],
                            [0, 1]
                        ],
```
I think in all cases in your current json files this will suffice as the reader. We also want the json module to raise whatever error is appropriate for the file given.
```suggestion
    with open(file_path) as json_file:
        contents_dict = json.load(json_file)

    for k, value in contents_dict.items():
        if isinstance(value, list) and k in ('time', 'data'):
            contents_dict[k] = np.asarray(value)
    return contents_dict
```